src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/sync/FlowSyncFsm.java

```diff
@@ -0,0 +1,303 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.fsm.sync;
+
+import org.openkilda.model.PathId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.wfm.CommandContext;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesManager;
+import org.openkilda.wfm.share.logger.FlowOperationsDashboardLogger;
+import org.openkilda.wfm.share.utils.FsmExecutor;
+import org.openkilda.wfm.topology.flowhs.exception.UnknownKeyException;
+import org.openkilda.wfm.topology.flowhs.fsm.FsmUtil;
+import org.openkilda.wfm.topology.flowhs.fsm.common.FlowPathSwappingFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.FlowSyncFsm.Event;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.FlowSyncFsm.State;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.actions.CreateSyncHandlersAction;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.actions.FailedCompleteAction;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.actions.FlowSyncSetupAction;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.actions.PathOperationResponseAction;
+import org.openkilda.wfm.topology.flowhs.fsm.sync.actions.SuccessCompleteAction;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathOperationConfig;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathOperationDescriptor;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathResultCode;
+import org.openkilda.wfm.topology.flowhs.service.FlowGenericCarrier;
+import org.openkilda.wfm.topology.flowhs.service.FlowProcessingEventListener;
+import org.openkilda.wfm.topology.flowhs.service.FlowSyncCarrier;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.squirrelframework.foundation.fsm.StateMachineBuilder;
+import org.squirrelframework.foundation.fsm.StateMachineBuilderFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+// TODO(surabujin): reconsider parent class (we do not use most part of it's tools)
+@Slf4j
+public class FlowSyncFsm
```

Comment:
Do we need to add a doc for this FSM?

Reply:
Yes - FSM state diagramm have been added.